10-Reflection/Reflection.Tasks/CommonTasks.cs

```diff
@@ -60,8 +64,25 @@ public static class CommonTasks
         /// <param name="propertyPath">dot-separated property path</param>
         /// <param name="value">assigned value</param>
         public static void SetPropertyValue(this object obj, string propertyPath, object value) {
-            // TODO : Implement SetPropertyValue method
-            throw new NotImplementedException();
+            var properties = propertyPath.Split('.');
+            foreach (var property in properties)
+            {
+                if (property != properties.Last())
+                {
+                    obj = obj.GetType().GetProperty(property).GetValue(obj);
+                }
+                else
+                {
+                    if (obj.GetType().GetProperty(property).CanWrite)
```

Comment:
make solution more generic so that it works for any number of layers of inheritance